src/plugins/procmon/winnt.cpp

```diff
@@ -0,0 +1,168 @@
+ * tamas.k.lengyel@gmail.com for further information.                      *
+ *                                                                         *
+ * If you have received a written license agreement or contract for        *
+ * Covered Software stating terms other than these, you may choose to use  *
+ * and redistribute Covered Software under those terms instead of these.   *
+ *                                                                         *
+ * Source is provided to this software because we believe users have a     *
+ * right to know exactly what a program is going to do before they run it. *
+ * This also allows you to audit the software for security holes.          *
+ *                                                                         *
+ * Source code also allows you to port DRAKVUF to new platforms, fix bugs, *
+ * and add new features.  You are highly encouraged to submit your changes *
+ * on https://github.com/tklengyel/drakvuf, or by other methods.           *
+ * By sending these changes, it is understood (unless you specify          *
+ * otherwise) that you are offering unlimited, non-exclusive right to      *
+ * reuse, modify, and relicense the code.  DRAKVUF will always be          *
+ * available Open Source, but this is important because the inability to   *
+ * relicense code has caused devastating problems for other Free Software  *
+ * projects (such as KDE and NASM).                                        *
+ * To specify special license conditions of your contributions, just say   *
+ * so when you send them.                                                  *
+ *                                                                         *
+ * This program is distributed in the hope that it will be useful, but     *
+ * WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+ * license file for more details (it's in a COPYING file included with     *
+ * DRAKVUF, and also available from                                        *
+ * https://github.com/tklengyel/drakvuf/COPYING)                           *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "winnt.h"
+#include <stdio.h>
+
+void print_protection_attributes(uint32_t attributes, char sep)
+{
+    if (attributes == 0)
+    {
+        printf("NONE");
+        return;
+    }
+
+    int num = 0;
+    for (size_t i = 0; i < 8 * sizeof(attributes) && attributes; ++i)
+    {
+        uint32_t attr = attributes & (1 << i);
+        if (attr == 0)
+            continue;
+
+        attributes &= ~attr;
+        ++num;
+        if (num > 1)
+            printf("%c", sep);
+
+        switch (attr)
+        {
+            case PAGE_NOACCESS:
+                printf("PAGE_NOACCESS");
```

Comment:
I don't like how this function prints strings by itself. I would prefer if this function just returned a `const char*` once to the caller and then the caller prints it.